tests/jest/components/InventoryReportRequestModal.test.tsx

```diff
@@ -0,0 +1,674 @@
+    });
+
+    it("returns the expected value for generate report (post) requests", async () => {
+      const result = await AdminAPI.requestInventoryReport(API_ENDPOINT_PARAMS);
+      expect(result).toEqual(GENERATE_REPORT_SUCCESS);
+    });
+    it("returns an error value for unsuccessful generate report (post) requests", async () => {
+      server.use(
+        rest.post(MOCK_SERVER_API_ENDPOINT_PATH, (req, res, ctx) =>
+          res(ctx.status(404))
+        )
+      );
+      let error = undefined;
+      try {
+        await AdminAPI.requestInventoryReport(API_ENDPOINT_PARAMS);
+      } catch (e) {
+        error = await e;
+      }
+      expect(error?.message).toMatch("Request failed with status 404: POST ");
+    });
+  });
+
+  describe("component rendering", () => {
+    const onHide = jest.fn();
+    const server = setupMockServer();
+
+    beforeAll(() => server.listen());
+    afterAll(() => server.close());
+    afterEach(() => {
+      server.resetHandlers();
+      server.restoreHandlers();
+      onHide.mockReset();
+      queryClient.clear();
+    });
+
+    const LIBRARY = "library-short-name";
+    const email = "librarian@example.com";
+    const EMAIL_CONTEXT_PROVIDER_PROPS = { email };
+    const NO_EMAIL_CONTEXT_PROVIDER_PROPS = { email: undefined };
+
+    it("hides the modal when `show` is false", async () => {
+      const {
+        container,
+        queryAllByRole,
+      } = renderWithProviders(
+        <InventoryReportRequestModal
+          show={false}
+          onHide={onHide}
+          library={LIBRARY}
+        />,
+        { queryClient }
+      );
+      expect(container.querySelectorAll(".modal-content")).toHaveLength(0);
+      expect(
+        queryAllByRole("button", { name: CONFIRM_BUTTON_CONTENT })
+      ).toHaveLength(0);
+      expect(
+        queryAllByRole("button", { name: CANCEL_BUTTON_CONTENT })
+      ).toHaveLength(0);
+    });
+
+    it("displays the modal when `show` is true", async () => {
+      const {
+        container,
+        getByRole,
+      } = renderWithProviders(
+        <InventoryReportRequestModal
+          show={true}
+          onHide={onHide}
+          library={LIBRARY}
+        />,
+        { queryClient }
+      );
+      container.querySelector(".modal-content");
+      const modalContent = getByRole("document");
+      const heading = within(modalContent).getByRole("heading", {
+        level: MODAL_HEADING_LEVEL,
+      });
+      expect(heading).toHaveTextContent(CONFIRMATION_MODAL_HEADING);
+    });
+
+    it("initially shows a request confirmation", () => {
+      const {
+        container,
+        getByRole,
+      } = renderWithProviders(
+        <InventoryReportRequestModal
+          show={true}
+          onHide={onHide}
+          library={LIBRARY}
+        />,
+        { queryClient }
+      );
+      container.querySelector(".modal-content");
+      const modalContent = getByRole("document");
```

Comment:
It feels like `getByRole("dialog")` should be in here somewhere, since that would be the distinguishing characteristic of what you're looking for. But I don't know if react-bootstrap is setting that role.

Reply:
I did try using that initially, but ended up with an object with which I couldn't do anything else, so I took a different tack. I did manage to sort out some of the other problems I was having, so I'll take another look at it.

Reply:
The issue is that there are two elements with the "dialog" role, which I had somehow missed before. I switched to `getAllByRole` and `queryAllByRole` to resolve this and streamlined the tests in the process.